### PR TITLE
Initial changes for prod-dns-to-iac

### PR DESCRIPTION
**Pull request by MisterSK**
### JIRA link (if applicable) ###
https://dev.azure.com/hmcts/Platform%20Engineering/_workitems/edit/660

### Change description ###
The changes add the necessary TF infrastructure files for production dns zones and recordsets

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: components/prod/azure-public-dns.tf [hunk 1]
```diff
@@ -0,0 +1,175 @@
+module "public-dns-1" {
```
Comment (timja) [inline components/prod/azure-public-dns.tf]: this should have the name of the zone in it

File: components/prod/azure-public-dns.tf [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  source              = "../../modules/azure-public-dns/"
+  cname_records       = yamldecode(data.local_file.configuration.content).cname
+  ns_recordsets       = yamldecode(data.local_file.configuration.content).ns
+  mx_recordsets       = yamldecode(data.local_file.configuration.content).mx
+  a_recordsets        = yamldecode(data.local_file.configuration.content).A
+  txt_recordsets      = yamldecode(data.local_file.configuration.content).txt
+  zone_name           = zone1
```
Comment (timja) [inline components/prod/azure-public-dns.tf]: name of zone should be here

File: environments/prod/apply-divorce.service.gov.uk.tfvars [hunk 3]
```diff
@@ -0,0 +1,33 @@
+recordsets = [
```
Comment (timja) [inline environments/prod/apply-divorce.service.gov.uk.tfvars]: this needs formatting

File: components/prod/azure-public-dns.tf (hunk 1 above)
Comment (MisterSK) [reply components/prod/azure-public-dns.tf]: Had chat with @naikajah about this before this initial PR - will implement addition of zone names and update

File: components/prod/azure-public-dns.tf (hunk 2 above)
Comment (MisterSK) [reply components/prod/azure-public-dns.tf]: Same as above...

File: components/prod/azure-public-dns.tf (hunk 1 above)
Comment (MisterSK) [reply components/prod/azure-public-dns.tf]: @timja - agree with the stepped PR, 1 zone at a time. Are we going to be merging each zone addition to master as I believe we won't get the TF plan confirmation "Nothing to change" until all zones/recordsets are imported

Comment (timja) [reply components/prod/azure-public-dns.tf]: yes we should be merging each one to master 1 at a time, I see no reason we will see no change as long as we import each zone at a time?

**Comment by timja** [comment]
Closing, as we've taken the approach of one at a time

**Pull request closed without merging**